### PR TITLE
Replace PIL renderer with Reportlab renderer

### PR DESCRIPTION
**Pull request by nbessi**
It allows to have lighter, faster, vectorial payment slip.

This should fix most of current performance and OCR issues

### CODE REVIEW

**Comment by PCatinean** [comment]
:+1:  Works nicely even with standard Ubuntu python-reportlab packages

**Comment by nbessi** [comment]
Maybe we should add a migration step that reset vertical and horizontal delta

**Comment by PCatinean** [comment]
That's a good idea, since quite a few people might be migrating

**Comment by yvaucher** [comment]
:+1: Would be great to add migration step to convert those mm in inches.

**Comment by fclementic2c** [comment]
:+1:

File: l10n_ch_payment_slip/i18n/fr.po [hunk 1]
```diff
@@ -23,22 +23,22 @@ msgstr "Données BVR"
 
 #. module: l10n_ch_payment_slip
 #: field:res.company,bvr_delta_horz:0
-msgid "BVR Horz. Delta (px)"
+msgid "BVR Horz. Delta (inch)"
 msgstr "Décalage horizontal global en pixels"
```
Comment (guewen) [inline l10n_ch_payment_slip/i18n/fr.po]: The translation still says "en pixels", should be "en pouces" (same comment for the other translations in the file)

**Comment by lsimonin** [comment]
Do you have any idea when the pull request will be validated?

**Comment by nbessi** [comment]
Branche rebased and french translation corrected

**Comment by nbessi** [comment]
@lsimonin I guess it can be merged now except, if a reviewer insists to have a migration script in this PR

**Comment by lsimonin** [comment]
@nbessi Thank's for you response. I  waiting for the merge.

**Comment by mdietrichc2c** [comment]
:+1:

**Comment by lsimonin** [comment]
Can you make a patch or merge this PR? i would'like to test this.

**Comment by nbessi** [comment]
@lsimonin you can try directly without waiting it to be merged ??

**Comment by lsimonin** [comment]
how can I recover all the files modified once without having modified each line? Is it possible to recover all the changes at once?

**Comment by nbessi** [comment]
@lsimonin  you should have a look at `git remote add` and `git checkout` commands

**Comment by guewen** [comment]
@lsimonin 

```
git fetch origin
git pull origin pull/105/head
```

**Comment by lsimonin** [comment]
@guewen Thank you it works fine.
When i print the BVR, I still have quality problems throughout the bvr. Is this normal?

image

**Comment by nbessi** [comment]
@guewen @lsimonin No it seems that the modification has not been taken in account. @PCatinean and @fclementic2c can confirm it works

**Comment by lsimonin** [comment]
Now it works. ;) But i have 3 remarks
1. this sentence is not translated: "Payment slip related to invoice FC2015/0003 due on the 17/03/2015"
2. The company's address is not displayed on the bvr
   image
3. it would be nice to have when viewing documents the possibility of the following reports:
4. single invoice
5. BVR without background (for printing)
6. Invoice + bvr with background (for sending invoices by email)

**Comment by nbessi** [comment]
@lsimonin all this is possible to choose what to print and how to print it you have to modify your setting in company and in the bank account.

**Comment by lsimonin** [comment]
@nbessi Ok thanks for the adress.it works
For the background it is possible that they send 80% of invoices by email and some paper. the aim being to avoid the user to edit the report.
Also when sending invoice by email, it is not possible to join 2 report.

**Comment by fclementic2c** [comment]
@lsimonin Thanks for you review. Did you notice that you have several options you can select: 
- on the bank form allowing you to print or not the bank, the account number, the partner adress
  like where you can dicide to display or not the background, adresses... ?
  image
- on the company form (tab BVR data) : here you can insert the background if you like 
  image

Frederic

**Comment by lsimonin** [comment]
@fclementic2c Thank you response

On the tab bvr data i see this. but for example i send 80% of invoices by email and some times by paper. the aim being to avoid the user to edit the configuration
Also when sending invoice by email, it is not possible to join 2 report. (impossible to attach the invoice and bvr)

**Comment by nbessi** [comment]
@lsimonin to merge report we use https://github.com/OCA/reporting-engine/tree/7.0/base_report_assembler but it seems that it is not ported yet in version 8.0.

For your other concerns you should open an enchancement issue. They are out of the scope of this PR.

**Comment by nbessi** [comment]
@lsimonin I will add the missing translation

**Comment by yvaucher** [comment]
Just add need fixing so we don't forget the missing translation

**Comment by rdeheele** [comment]
:+1:

**Comment by lsimonin** [comment]
I had a return of the validation of the BVR by Postfinance.
It miss the "Reference number" of the BVR in the "Récépissé"
Please fix this.

**Comment by lsimonin** [comment]
Can you please increase the space between "Payment slip to invoice ..." and the BVR

Please see this.
paymentslip

**Comment by lsimonin** [comment]
Can you please increase the space between "Payment slip to invoice ..." and the BVR

Please see this.
paymentslip

**Comment by nbessi** [comment]
@lsimonin Fixed

**Comment by lsimonin** [comment]
When I move the address, reference number at the top does not move at the same time in the "Récépissé"

**Comment by lsimonin** [comment]
See the problem with my last message. See the red line.
The first 2 character are in the margin and are not printed.
bvr

**Comment by lsimonin** [comment]
the bvr was validated by Postfinance

**Comment by nbessi** [comment]
@lsimonin That's a good new I will fix your last comment ASAP

**Comment by nbessi** [comment]
Positioning fixed

**Comment by yvaucher** [comment]
Ok let's merge as it was validated by PostFinance

**Pull request merged**